src/AutoRest.CSharp/Mgmt/Decorator/SchemaExtensions.cs

```diff
@@ -75,7 +75,7 @@ public static bool IsResourceModel(this Schema schema)
             switch (property.SerializedName)
             {
                 case "id":
-                    if (property.Schema.Type == AllSchemaTypes.String)
+                    if (property.Schema.Type == AllSchemaTypes.String || property.Schema.Type == AllSchemaTypes.ArmId)
```

Comment:
Please add test for this change

Reply:
Good point, updated the existing test to verify this scenario.